### PR TITLE
Feature/9494 qr scanner view model tests

### PR DESCRIPTION
**Pull request by 30mar**
## Description
Add unit tests for the QRScannerViewModel

## Link to Jira
https://jira-ibs.wbs.net.sap/browse/EXPOSUREAPP-9494

### CODE REVIEW

File: src/xcode/ENA/ENA/Source/Scenes/QRScanner/_tests_/QRScannerViewModelTests.swift [hunk 1]
```diff
@@ -0,0 +1,286 @@
...
+			case .success(let result):
+				switch result {
+				case .certificate(let person, let certificate):
+					XCTAssertEqual(person.dateOfBirth, "1981-08-30", "Expected correct person dateOfBirth")
+					XCTAssertEqual(person.name?.standardizedName, "MITCHELL ROBERT", "Expected correct person standardizedName")
+					XCTAssertEqual(certificate.type, .vaccination, "Expected correct person standardizedName")
+					onFailureExpectation.fulfill()
+				default:
+					XCTFail("Expected a successful scan of Vaccination Certificate")
+				}
```
Comment (nickguendling) [inline src/xcode/ENA/ENA/Source/Scenes/QRScanner/_tests_/QRScannerViewModelTests.swift:267]:
```suggestion
				XCTFail("Expected scan to fail")
```

This should not succeed, right?

File: src/xcode/ENA/ENA/Source/Scenes/QRScanner/QRScannerViewModel.swift [hunk 2]
```diff
@@ -47,7 +47,8 @@ class QRScannerViewModel: NSObject, AVCaptureMetadataOutputObjectsDelegate {
 	
 	func didScan(metadataObjects: [MetadataObject]) {
 		guard isScanningActivated else {
-			Log.info("Scanning not stopped from previous run")
+			completion(.failure(.scanningError(.scanningDeactivated)))
```
Comment (nickguendling) [inline src/xcode/ENA/ENA/Source/Scenes/QRScanner/QRScannerViewModel.swift:50]: Is there anything we can tell the user if this happens? Wouldn't it be a coding error if this happens? Not sure if it makes sense to return an error here 🤔

File: src/xcode/ENA/ENA/Source/Scenes/QRScanner/QRScannerViewModel.swift [hunk 3]
```diff
@@ -186,7 +188,8 @@ class QRScannerViewModel: NSObject, AVCaptureMetadataOutputObjectsDelegate {
 		}
 		
 		guard let parser = parser else {
-			Log.error("QRCode parser not intitialized, Scanned code prefix doesnt match any of the scannable structs", log: .qrCode, error: nil)
+			Log.error("QRCode parser not initialized, Scanned code prefix doesn't match any of the scannable structs", log: .qrCode, error: nil)
+			self.completion(.failure(.scanningError(.codeNotFound)))
```
Comment (nickguendling) [inline src/xcode/ENA/ENA/Source/Scenes/QRScanner/QRScannerViewModel.swift:192]:
```suggestion
			completion(.failure(.scanningError(.codeNotFound)))
```

**Review by nickguendling — CHANGES_REQUESTED** [review]

File: src/xcode/ENA/ENA/Source/Scenes/QRScanner/_tests_/QRScannerViewModelTests.swift (hunk 1 above)
Comment (30mar) [reply src/xcode/ENA/ENA/Source/Scenes/QRScanner/_tests_/QRScannerViewModelTests.swift:267]: yeah correct

File: src/xcode/ENA/ENA/Source/Scenes/QRScanner/QRScannerViewModel.swift (hunk 2 above)
Comment (nickguendling) [reply src/xcode/ENA/ENA/Source/Scenes/QRScanner/QRScannerViewModel.swift:50]: Okay, seems fine for me to show the same error message

**Review by nickguendling — APPROVED** [review]

**Review by naveeddotio — APPROVED** [review]

**Pull request merged**